psydac/cad/geometry.py

```diff
@@ -326,3 +327,216 @@ def export( self, filename ):
+
+    # Write geometry metadata as fixed-length array of ASCII characters
+    h5['geometry.yml'] = np.array( geom, dtype='S' )
+    # ...
+
+    # ... topology
+    if patch.dim == 1:
+        bounds1 = (patch.breaks(0)[0], patch.breaks(0)[-1])
+        domain  = Line(patch_name, bounds1=bounds1)
+
+    elif patch.dim == 2:
+        bounds1 = (patch.breaks(0)[0], patch.breaks(0)[-1])
+        bounds2 = (patch.breaks(1)[0], patch.breaks(1)[-1])
+        domain  = Square(patch_name, bounds1=bounds1, bounds2=bounds2)
+
+    elif patch.dim == 3:
+        bounds1 = (patch.breaks(0)[0], patch.breaks(0)[-1])
+        bounds2 = (patch.breaks(1)[0], patch.breaks(1)[-1])
+        bounds3 = (patch.breaks(2)[0], patch.breaks(2)[-1])
+        domain  = Cube(patch_name, bounds1=bounds1, bounds2=bounds2, bounds3=bounds3)
+
+    topo_yml = domain.todict()
+
+    # Dump geometry metadata to string in YAML file format
+    geom = yaml.dump( data   = topo_yml,
+                     Dumper = yamlloader.ordereddict.Dumper )
+    # Write topology metadata as fixed-length array of ASCII characters
+    h5['topology.yml'] = np.array( geom, dtype='S' )
+
+    group = h5.create_group( yml['patches'][i]['mapping_id'] )
+    group.attrs['degree'     ] = patch.degree
+    group.attrs['rational'   ] = rational
+    group.attrs['periodic'   ] = tuple( False for d in range( patch.dim ) ) if periodic is None else periodic
+    for d in range( patch.dim ):
+        group['knots_{}'.format( d )] = patch.knots[d]
+    group['points'] = patch.points
+    if rational:
+        group['weights'] = patch.weights
+
+    h5.close()
+
+#==============================================================================
+def refine(nrb, ncells=None, degree=None , periodic=None):
+
+    nrb = nrb.clone()
+    if ncells is not None:
+
+        for axis in range(0,nrb.dim):
+            ub = nrb.knots[axis][0]
+            ue = nrb.knots[axis][-1]
+            knots = list(np.linspace(ub,ue,ncells[axis]+2)[1:-1])
+            for k in nrb.knots[axis]:
+                if k in knots:
+                    knots.remove(k)
+            nrb.refine(axis, knots)
+
+    if degree is not None:
+        for axis in range(0,nrb.dim):
+            d = degree[axis] - nrb.degree[axis]
+            nrb.elevate(axis, times=d)
+
+    return nrb
+
+#==============================================================================
+def create_geometry_file(nrb, filename, ncells, degree, periodic=None):
+
+    import igakit
+    assert isinstance(nrb, igakit.nurbs.NURBS)
+    import os.path
+
+    nrb = refine(nrb, ncells=[i-1 for i in ncells], degree=degree)
+    # ...
+
+    extension = os.path.splitext(filename)[1]
+    if not extension == '.h5':
+        raise ValueError('> Only h5 extension is allowed for filename')
+
+    export_geo(filename, nrb, periodic=periodic)
+
+#==============================================================================
+class geopdes(object):
```

Comment:
1. Use capitalized names for classes, for example `GeoPDEs` in this case.
2. Please add a docstring that explains what this class does
3. If I understand correctly the only goal of this class is to convert a GeoPDEs file to a IgaKit NURBS object, but do you actually need multiple instances of this "converter"? If not, then you do not need a class at all, and you can just write a single function...

Reply:
yes you are right I will just convert it to a function

Reply:
Thanks